vault/external_tests/identity/login_mfa_totp_test.go

```diff
@@ -191,26 +196,37 @@ func TestLoginMfaGenerateTOTPTestAuditIncluded(t *testing.T) {
 	}
 
 	// validation
-	time.Sleep(5 * time.Second)
 	var totpPasscode1 string
-	testhelpers.RetryUntil(t, 20*time.Second, func() error {
+	validateFunc := func(methodIdentifier string) {
```

Comment:
You're confident that we can avoid retrying, that this change won't make the test flaky?

Reply:
Actually, it feels a bit odd to have a retry for this because, for one, if it takes 20 seconds to have a new code when the totp code generation period is 5 seconds, then we are masking an error that we probably should know or even fix. Two, this wait time is deterministic, meaning that totp engine should be generating a new code after 5 seconds, and normally I think `RetryUntil` is used for randomized behaviour.

Reply:
What if we get the code from the totp mount, and 1ms after that the code changes?

Reply:
That is fair. And, I think this test is not meant to test totp engine code generation. I will revert my changes to use `RetryUntil`.

Reply:
Ah, I just realized that there was another reason as well that I needed to use the sleep here. So, in code, we only allow a maximum number of consecutive TOTP validation failures. And, without the sleep, we keep trying and then we hit `maximum TOTP validation attempts %d exceeded the allowed attempts...` failure. So, `RetryUntil` does not work as is since the sleep time in that 100 milliseconds. I am going to modify that to take `sleepTime` duration as a new parameter.